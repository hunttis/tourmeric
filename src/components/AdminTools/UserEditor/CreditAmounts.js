import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import { isLoaded } from 'react-redux-firebase';
import _ from 'lodash';
import { Translate } from 'react-localize-redux';

import { mapCategoryToColor } from '../../Common/Utils';

export const CreditAmounts = ({ userId, storecredit, storecreditcategories }) => (
  <Fragment>
    {/* <div>FFuuuu {userId}</div> */}
    {isLoaded(storecreditcategories) &&
      <Fragment>
        {Object.entries(storecreditcategories).map((categoryEntry) => {
          const categoryKey = categoryEntry[0];
          const categoryValue = categoryEntry[1];
          const userData = storecredit[userId];
          const dataForCategory = _.filter(userData, (data) => {
            if (categoryKey === 'white') {
              return data.category === categoryKey || !data.category || data.category === '';
            }
            return data.category === categoryKey;
          });
          const categorySum = _.reduce(dataForCategory, (totalForCategory, data) => totalForCategory + parseFloat(data.value), 0);
          return (
            <li key={`${userId}-${categoryKey}`}><span className={`has-text-${mapCategoryToColor(categoryKey)}`}>{categoryValue}</span> : {categorySum.toFixed(2)} €</li>
          );
        })}
        <li>&nbsp;</li>
        <li className="is-all-caps">
          <strong><Translate id="total" />: {_.reduce(storecredit[userId], (totalForUser, data) => totalForUser + parseFloat(data.value), 0)} €</strong>
        </li>
      </Fragment>
    }
  </Fragment>
);

CreditAmounts.propTypes = {
  userId: PropTypes.string.isRequired,
  storecredit: PropTypes.object.isRequired,
  storecreditcategories: PropTypes.object.isRequired,
};
