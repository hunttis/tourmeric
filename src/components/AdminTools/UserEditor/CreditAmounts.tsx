import React from 'react';
import { isLoaded } from 'react-redux-firebase';
import _ from 'lodash';
import { FormattedMessage, IntlShape } from "react-intl";

import { mapCategoryToColor } from '../../Common/Utils';

interface CreditAmountsProps {
  userId: string;
  storecredit: { [key: string]: { [key: string]: StoreCredit } };
  storecreditcategories: { [key: string]: string };
}

interface StoreCredit {
  category: string;
  value: number;
}

export const CreditAmounts = ({ userId, storecredit, storecreditcategories }: CreditAmountsProps) => (
  <>
    {isLoaded(storecreditcategories) &&
      <>
        {Object.entries(storecreditcategories).map((categoryEntry) => {
          const categoryKey = categoryEntry[0];
          const categoryValue = categoryEntry[1];
          const userData = storecredit[userId];
          const dataForCategory = _.filter(userData, (data) => {
            if (categoryKey === 'white') {
              return data.category === categoryKey || !data.category || data.category === '';
            }
            return data.category === categoryKey;
          });
          const categorySum = _.reduce(dataForCategory, (totalForCategory, data) => totalForCategory + data.value, 0.0).toFixed(2);
          return (
            <li key={`${userId}-${categoryKey}`}><span className={`has-text-${mapCategoryToColor(categoryKey)}`}>{categoryValue}</span> : {categorySum} €</li>
          );
        })}
        <li>&nbsp;</li>
        <li className="is-all-caps">
          <strong><FormattedMessage id="total" />: {_.reduce(storecredit[userId], (totalForUser, data) => totalForUser + data.value, 0).toFixed(2)} €</strong>
        </li>
      </>
    }
    {!isLoaded(storecreditcategories) &&
      <div>Loading</div>
    }
  </>
);
