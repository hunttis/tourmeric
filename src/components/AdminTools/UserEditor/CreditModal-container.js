import { connect } from 'react-redux';
import { compose } from 'redux';
import { firebaseConnect } from 'react-redux-firebase';

import CreditModal from './CreditModal';

export default compose(
  firebaseConnect([
    { path: '/storecredit' },
  ]),
  connect(state => ({
    storecredit: state.firebase.data.storecredit,
  })),
  connect(({ firebase: { auth, profile } }) => ({ auth, profile })),
)(CreditModal);
