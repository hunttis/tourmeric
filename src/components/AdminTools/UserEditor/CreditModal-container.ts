import { connect } from 'react-redux';
import { compose } from 'redux';

import CreditModal from './CreditModal';
import { ReduxState } from '../../../models/ReduxState';

export default compose(
  connect((state: ReduxState) => ({
    storecredit: state.firebase.data.storecredit,
    storecreditcategories: state.firebase.data.storecreditcategories,
  })),
  connect(({ firebase: { auth, profile } }: ReduxState) => ({ auth, profile })),
)(CreditModal) as React.ComponentType<any>;
