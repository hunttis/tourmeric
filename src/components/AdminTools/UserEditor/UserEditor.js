import React, { Component, Fragment } from 'react';
import { isLoaded } from 'react-redux-firebase';
import PropTypes from 'prop-types';
import _ from 'lodash';
import { Translate } from 'react-localize-redux';
import firebase from 'firebase/app';
import UserEntry from './UserEntry-container';
import EditableField from '../../Common/EditableField-container';
import StoreCreditTable from '../StoreCredit/StoreCreditTable-container';

export default class UserEditor extends Component {

  constructor(props) {
    super(props);
    this.searchUsers.bind(this.searchUsers);
    this.escFunction = this.escFunction.bind(this);
  }

  state = {
    filteredUsers: [],
    modalOpenClass: '',
    modalUser: '',
    modalMode: '',
    creditFormAmount: 0.0,
    creditFormNote: '',
  }

  componentDidMount() {
    document.addEventListener('keydown', this.escFunction, false);
  }

  componentWillUnmount() {
    document.removeEventListener('keydown', this.escFunction, false);
  }

  getUser(userid) {
    if (!userid) {
      return null;
    }
    const { users } = this.props;
    const foundUser = _.find(users, { key: userid });
    return foundUser;
  }

  escFunction(event) {
    if (event.keyCode === 27) {
      this.closeModal();
    }
  }

  searchUsers(event) {
    const { users } = this.props;
    const searchString = event.target.value;
    if (_.isEmpty(searchString)) {
      this.setState({ filteredUsers: [] });
    } else {
      const filtered = Object.values(users).filter((userEntry) => {
        const user = userEntry.value;
        return (user.email && user.email.toLowerCase().indexOf(searchString) !== -1) ||
          (user.displayName && user.displayName.toLowerCase().indexOf(searchString) !== -1) ||
          (user.username && user.username.toLowerCase().indexOf(searchString) !== -1);
      });
      this.setState({ filteredUsers: filtered });
    }
  }

  openEditModal(userid) {
    this.setState({ modalUser: userid, modalMode: 'edit', modalOpenClass: 'is-active' });
  }

  openCreditModal(userid) {
    this.setState({ modalUser: userid, modalMode: 'credit', modalOpenClass: 'is-active' });
  }

  openDisableModal(userid) {
    this.setState({ modalUser: userid, modalMode: 'disable', modalOpenClass: 'is-active' });
  }

  closeModal() {
    this.setState({ modalUser: '', modalMode: '', modalOpenClass: '', creditFormAmount: '', creditFormNote: '' });
  }

  disableUser(userId) {
    firebase.update(`/users/${userId}`, { active: false });
  }

  calculateTotal(creditData) {
    let total = 0.0;
    if (creditData[1]) {
      for (const dataItem of Object.values(creditData[1])) {
        total += dataItem.value;
      }
    }
    return total;
  }

  changeCreditNote(event) {
    this.setState({ creditFormNote: event.target.value });
  }

  changeCreditAmount(event) {
    this.setState({ creditFormAmount: event.target.value });
  }

  saveCredit(userId, note, amount) {
    const storedObject = { creditAddedBy: firebase.auth().currentUser.uid, date: new Date().toUTCString(), note, value: Number.parseFloat(amount) };
    firebase.push(`/storecredit/${userId}`, storedObject);
    this.setState({ creditFormNote: '', creditFormAmount: 0.0 });
  }

  renderEditModal(userId, userData) {
    return (
      <Fragment>
        <div className="box">
          <EditableField
            defaultValue={userData.displayName}
            labelContent="shownname"
            placeHolder="exampleshownname"
            path={`/users/${userId}`}
            targetName="displayName"
          />

          <EditableField
            defaultValue={userData.email}
            labelContent="email"
            placeHolder="exampleemail"
            path={`/users/${userId}`}
            targetName="email"
          />

          <EditableField
            defaultValue={userData.avatarUrl}
            labelContent="avatarurl"
            placeHolder="avatar image url"
            path={`/users/${userId}`}
            targetName="avatarURL"
          />

          <EditableField
            defaultValue={userData.username}
            labelContent="username"
            placeHolder="username"
            path={`/users/${userId}`}
            targetName="username"
          />

          <div className="field is-horizontal">
            <div className="field-label is-normal">
              <label className="label"><Translate id="role" /></label>
            </div>
            <div className="field-body">
              <div className="field">
                <div className="control is-expanded">
                  <div className="select">
                    <select
                      defaultValue={userData.role}
                      onChange={event => firebase.update(`/users/${userId}`, { role: event.target.value })}
                    >
                      <option value={null}><Translate id="select" /></option>
                      <option value="user"><Translate id="roleuser" /></option>
                      <option value="admin"><Translate id="roleadmin" /></option>
                    </select>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </Fragment>
    );
  }

  renderCreditModal(userId, storecredit) {
    const userCreditData = storecredit[userId];
    return (
      <Fragment>
        {userCreditData &&
          <StoreCreditTable key={userId} userId={userId} creditData={userCreditData} />
        }
        <div className="level">

          <div className="level-left">
            <div className="field">
              <label className="label">Note</label>
              <input className="input" type="text" defaultValue={this.state.creditFormNote} placeholder="Credit change note" onChange={event => this.changeCreditNote(event)} />
            </div>
          </div>
          <div className="level-item">
            <div className="field">
              <label className="label">Credit Amount</label>
              <input className="input" type="number" defaultValue={this.state.creditFormAmount} placeholder="Credit amount" onChange={event => this.changeCreditAmount(event)} />
            </div>
          </div>
          <div>
            <button className="button is-primary" onClick={() => this.saveCredit(this.state.modalUser, this.state.creditFormNote, this.state.creditFormAmount)}>Save</button>
          </div>
        </div>
      </Fragment>
    );
  }

  renderDisableModal(userId, userData) {
    return (
      <Fragment>

        <div className="box">
          <h1 className="title"><Translate id="disablethisuser" /></h1>
          <ul>
            {userData.displayName &&
              <li><Translate id="displayname" />: {userData.displayName}</li>
            }
            {userData.username &&
              <li><Translate id="username" />: {userData.username}</li>
            }
            <li><Translate id="email" />: {userData.email}</li>
            {userData.avatarUrl &&
              <li><Translate id="avatarurl" />:&nbsp;
                <a target="_blank" rel="noopener noreferrer" href={userData.avatarUrl}>{userData.avatarUrl.substr(0, 20)}...</a>
              </li>
            }
            {userData.dciNumber &&
              <li><Translate id="dcinumber" />: {userData.dciNumber}</li>
            }
          </ul>
          <hr />
          <div className="level">
            <div className="level-item">
              <button className="button is-danger is-small" onClick={() => this.disableUser(userId)}>Yes</button>
              <button className="button is-primary is-large">No</button>
            </div>
          </div>
        </div>
      </Fragment>
    );
  }

  render() {
    const { users, storecredit } = this.props;
    const { filteredUsers, modalOpenClass, modalUser, modalMode } = this.state;

    if (isLoaded(users) && isLoaded(storecredit)) {
      const currentUser = this.getUser(modalUser);
      const userId = currentUser ? currentUser.key : '';
      const userData = currentUser ? currentUser.value : {};
      const usedList = _.isEmpty(filteredUsers) ? Object.values(users) : filteredUsers;
      return (
        <Fragment>
          <div className={`modal ${modalOpenClass}`}>
            <div className="modal-background" onClick={() => this.closeModal()} />
            <div className="modal-content">
              {modalMode === 'edit' && this.renderEditModal(userId, userData)}
              {modalMode === 'credit' && this.renderCreditModal(userId, storecredit)}
              {modalMode === 'disable' && this.renderDisableModal(userId, userData)}
            </div>
            <button className="modal-close is-large" aria-label="close" onClick={() => this.closeModal()} />
          </div>
          <div className="level">
            <div className="level-left">
              <div className="level-item">
                <input className="input" type="text" placeholder="Search users" onChange={event => this.searchUsers(event)} />
              </div>
              <div className="level-item">
                {!_.isEmpty(filteredUsers) && `${filteredUsers.length} hits with search`}
                {_.isEmpty(filteredUsers) && `${users.length} users in total`}
              </div>
            </div>
            <div className="level-right">
              {/* <div className="button" onClick={this.minimizeOpenedUsers} ><Translate id="minimizeall" />
              </div> */}
            </div>
          </div>
          <div className="columns is-multiline">
            {usedList.map((userEntry) => {
              const result = Object.entries(storecredit).filter((storecreditEntry) => {
                const creditId = storecreditEntry[0];
                return creditId === userEntry.key;
              });
              const total = _.isEmpty(result) ? 0 : this.calculateTotal(result[0]);

              return (<UserEntry
                key={userEntry.key}
                openEditModal={() => this.openEditModal(userEntry.key)}
                openCreditModal={() => this.openCreditModal(userEntry.key)}
                openDisableModal={() => this.openDisableModal(userEntry.key)}
                userData={userEntry.value}
                creditAmount={Number.parseFloat(total)}
              />);
            }) };
          </div>
        </Fragment>
      );
    }
    return <div><Translate id="loading" /></div>;
  }
}

UserEditor.propTypes = {
  users: PropTypes.array,
  storecredit: PropTypes.object,
};
