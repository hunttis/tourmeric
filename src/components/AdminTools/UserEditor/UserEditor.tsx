import React, { Component } from 'react';
import { isLoaded } from 'react-redux-firebase';
import _ from 'lodash';
import { FormattedMessage, IntlShape } from "react-intl";
import UserEntry from './UserEntry-container';
import { DisableModal } from './DisableModal';
import { EditModal } from './EditModal';
import CreditModal from './CreditModal-container';
import { setListener } from '../../Common/DocumentUtils';
import { User } from '../../../models/ReduxState';
import { TourmericStoreCreditData } from '../../../models/StoreCredit';
import { parseStartingLettersFromUserLastnames } from '../../Common/UserUtils';

interface Props {
  users: [{ key: string, value: User }];
  storecredit: { [key: string]: { [key: string]: TourmericStoreCreditData } };
}

interface State {
  searchingWith: string;
  searchLetter: string;
  searchPhrase: string;
  modalOpenClass: string;
  modalUser: string;
  modalMode: string;
  showOnlyEnabledUsers: boolean;
}

export default class UserEditor extends Component<Props, State> {

  constructor(props: Props) {
    super(props);
    this.searchUsers.bind(this.searchUsers);
    this.escFunction = this.escFunction.bind(this);

    this.state = {
      searchingWith: '',
      searchLetter: '',
      searchPhrase: '',
      modalOpenClass: '',
      modalUser: '',
      modalMode: '',
      showOnlyEnabledUsers: true,
    };
  }

  componentDidMount = () => {
    setListener('keydown', this.escFunction);
  }

  static getDerivedStateFromProps = (nextProps: Props, prevState: State) => {
    if (prevState.searchingWith === '' && isLoaded(nextProps.users)) {
      const alphabet: string[] = parseStartingLettersFromUserLastnames(nextProps.users);
      return { searchingWith: 'letter', searchLetter: alphabet[0] };
    }
    return null;
  }

  getUser(userid: string) {
    if (!userid) {
      return null;
    }
    const { users } = this.props;
    const foundUser = _.find(users, { key: userid });
    return foundUser;
  }

  escFunction(event: KeyboardEvent) {
    if (event.keyCode === 27) {
      this.closeModal();
    }
  }

  searchByAlphabet(searchLetter: string) {

    if (_.isEmpty(searchLetter)) {
      this.resetSearches();
    }

    if (!_.isEmpty(searchLetter)) {
      this.setState({ searchingWith: 'letter', searchPhrase: '', searchLetter });
    }
  }

  searchUsers(event: React.ChangeEvent<HTMLInputElement>) {
    const searchPhrase = event.target.value;

    if (_.isEmpty(searchPhrase)) {
      this.resetSearches();
    }

    if (!_.isEmpty(searchPhrase)) {
      this.setState({ searchingWith: 'phrase', searchPhrase, searchLetter: '' });
    }
  }

  resetSearches() {
    const alphabet = parseStartingLettersFromUserLastnames(this.props.users);
    this.setState({ searchingWith: 'letter', searchPhrase: '', searchLetter: alphabet[0] });
  }

  filterUsers() {
    const { users } = this.props;
    const { searchPhrase, searchLetter, searchingWith, showOnlyEnabledUsers } = this.state;
    let filtered = Object.values(users);
    if (searchingWith === 'phrase') {
      filtered = Object.values(users).filter((userEntry) => {
        const user = userEntry.value;
        return (user.email && user.email.toLowerCase().indexOf(searchPhrase.toLowerCase()) !== -1) ||
          (user && user.displayName && user.displayName.toLowerCase().indexOf(searchPhrase.toLowerCase()) !== -1) ||
          (user && user.username && user.username.toLowerCase().indexOf(searchPhrase.toLowerCase()) !== -1) ||
          (user && user.firstName && user.firstName.toLowerCase().indexOf(searchPhrase.toLowerCase()) !== -1) ||
          (user && user.lastName && user.lastName.toLowerCase().indexOf(searchPhrase.toLowerCase()) !== -1);
      });
    }

    if (searchingWith === 'letter') {
      filtered = Object.values(users).filter((userEntry) => {
        const user = userEntry.value;
        return user.lastName && user.lastName.toLowerCase().startsWith(searchLetter.toLowerCase());
      });
    }

    if (showOnlyEnabledUsers) {
      filtered = filtered.filter((user) => user.value.active !== false);
    }

    return filtered;
  }

  openEditModal(userid: string) {
    this.setState({ modalUser: userid, modalMode: 'edit', modalOpenClass: 'is-active' });
  }

  openCreditModal(userid: string) {
    this.setState({ modalUser: userid, modalMode: 'credit', modalOpenClass: 'is-active' });
  }

  openDisableModal(userid: string) {
    this.setState({ modalUser: userid, modalMode: 'disable', modalOpenClass: 'is-active' });
  }

  closeModal() {
    this.setState({ modalUser: '', modalMode: '', modalOpenClass: '' });
  }

  filterInfoString() {
    const { searchPhrase, searchLetter, searchingWith } = this.state;

    if (searchingWith === 'letter') {
      return (
        <>
          <FormattedMessage id="searchinglastnamesstartingwith" />&nbsp;<span className="has-text-success">{searchLetter}</span>
        </>
      );
    }
    if (searchingWith === 'phrase') {
      return (
        <>
          <FormattedMessage id="searchingfirstnamelastnameemailbyphrase" />&nbsp;<span className="has-text-success">{searchPhrase}</span>
        </>
      );
    }
    return <FormattedMessage id="showingallusers" />;
  }

  render() {
    const { users, storecredit } = this.props;
    const { modalOpenClass, modalUser, modalMode } = this.state;

    if (!isLoaded(users) || !isLoaded(storecredit)) {
      return <><FormattedMessage id="loading" /></>;
    }

    const currentUser = this.getUser(modalUser);

    const userId = _.get(currentUser, 'key');
    const userData = _.get(currentUser, 'value');
    const userList = this.filterUsers();
    const alphabet = parseStartingLettersFromUserLastnames(users);

    return (
      <>
        <div className={`modal ${modalOpenClass}`}>
          <div className="modal-background" onClick={() => this.closeModal()} />
          <div className="modal-content">
            {modalMode === 'edit' && <EditModal userId={userId} userData={userData} />}
            {modalMode === 'credit' && <CreditModal userId={userId} />}
            {modalMode === 'disable' && <DisableModal userId={userId} userData={userData} />}
          </div>
          <button className="modal-close is-large" aria-label="close" onClick={() => this.closeModal()} />
        </div>
        <div className="level">

          <div className="level-left">
            <div className="field">
              <div className="field-label is-normal">
                <label className="label"><FormattedMessage id="filterlastname" /></label>
              </div>
              <div className="field-body">
                {alphabet.map((letter: string, index: number) => (

                  <button
                    onClick={() => this.searchByAlphabet(letter)}
                    className={`button is-grouped is-small ${this.state.searchLetter === letter && 'is-success'}`}
                    key={`letterSearch-${index}`}
                  >{letter}
                  </button>
                ))}
              </div>
            </div>
          </div>
          <div className="level-item">
            <div className="field">
              <div className="field-label has-text-left">
                <label className="label"><FormattedMessage id="textsearch" /></label>
              </div>
              <div className="field-body">
                <input className="input" type="text" onChange={(event) => this.searchUsers(event)} />
              </div>
            </div>
          </div>
          <div className="level-right">
            <>{users.length} <FormattedMessage id="usersintotal" /></>
            <br />
            {(userList.length !== users.length) && <>{userList.length} <FormattedMessage id="hitswithsearch" /></>}
          </div>
        </div>
        <h2 className="subtitle">
          {this.filterInfoString()}
        </h2>
        <div className="columns is-multiline">
          {userList.map((userEntry) => (<UserEntry
            key={userEntry.key}
            userId={userEntry.key}
            openEditModal={() => this.openEditModal(userEntry.key)}
            openCreditModal={() => this.openCreditModal(userEntry.key)}
            openDisableModal={() => this.openDisableModal(userEntry.key)}
            userData={userEntry.value}
          />))}
        </div>
      </>
    );

  }
}
