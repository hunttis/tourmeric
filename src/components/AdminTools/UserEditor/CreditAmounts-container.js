import { connect } from 'react-redux';
import { compose } from 'redux';

import { CreditAmounts } from './CreditAmounts';

export default compose(
  connect(state => ({
    storecredit: state.firebase.data.storecredit,
    storecreditcategories: state.firebase.data.storecreditcategories,
  })),
  connect(({ firebase: { auth, profile } }) => ({ auth, profile })),
)(CreditAmounts);
