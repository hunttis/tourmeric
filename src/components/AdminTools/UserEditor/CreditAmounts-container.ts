import { connect } from 'react-redux';
import { compose } from 'redux';

import { CreditAmounts } from './CreditAmounts';
import { ReduxState } from '../../../models/ReduxState';

export default compose(
  connect((state: ReduxState) => ({
    storecredit: state.firebase.data.storecredit,
    storecreditcategories: state.firebase.data.storecreditcategories,
  })),
  connect(({ firebase: { auth, profile } }: ReduxState) => ({ auth, profile })),
)(CreditAmounts) as React.ComponentType<any>;
