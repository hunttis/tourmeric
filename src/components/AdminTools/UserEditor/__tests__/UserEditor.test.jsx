import React from 'react';
import { shallow } from 'enzyme';
import UserEditor from '../UserEditor';
import '../../../../__mocks__/setupTests';

describe('UserEditor Tests', () => {

  it('Compares UserEditor to snapshot', () => {

    const mockUsers = [];

    const userEditor = shallow(<UserEditor
      users={mockUsers}
    />);

    expect(userEditor.exists()).toBe(true);
    expect(userEditor).toMatchSnapshot();

  });

  // TODO: Write separate tests for no users, one user, several users
  // TODO: Write tests for hitting the esc key.
  // TODO: Write tests for opening and closing the modals.
});
