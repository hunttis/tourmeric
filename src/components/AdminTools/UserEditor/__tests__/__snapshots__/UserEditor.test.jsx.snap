// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserEditor Tests Compares UserEditor to snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <UserEditor
    users={Array []}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Translate
        id="loading"
      />,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "id": "loading",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Translate
          id="loading"
        />,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "id": "loading",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
